jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/RemoveDataCaseFileInstanceCommand.java

```diff
@@ -59,7 +63,15 @@ public Void execute(Context context) {
         
         // apply authorization
         authorizationManager.checkDataAuthorization(caseFile.getCaseId(), caseFile, variableNames);
-        
+          
+        org.jbpm.process.instance.ProcessInstance pi = (org.jbpm.process.instance.ProcessInstance) ksession.getProcessInstance(processInstanceId);
+        VariableScope variableScope = (VariableScope) pi.getContextContainer().getDefaultContext(VariableScope.VARIABLE_SCOPE);
+        for (String name: variableNames) {
+            if (caseFile.getData(name) != null && variableScope.isReadOnly(VariableScope.CASE_FILE_PREFIX+name)) {
+                throw new VariableViolationException(pi.getId(), name,"variable is read only and cannot be removed");
```

Comment:
Maybe use the same message as in `VariableScopeInstance`?
`throw new VariableViolationException(getProcessInstance().getId(), name, "Variable '" + name + "' is already set and is marked as read only");`

Reply:
Changed